[fix] #38 dehumanize-date, +packages, +npm@5

JeromeFitz opened the pull request:
🚧️ #38 dehumanize-date
This broke the buuild (seemingly) due to the package.json needing to be
updated to the latest.
At least, that is what had all the tests pass and the build pass.

🐻️ npm@5
+ package-lock.json
+ change `prepublish` to `prepare`
-- 📓️ Note: `prepublishOnly` = `upload-only` if that is desired
behaviour

📦️ package.json
+ Removed `peerDependencies` and put `react` into `dependencies`
-- 📓️ Note: It was messing up the pretest + test

📦️ Packages Updated
babel-jest@20.0.3
dehumanize-date@2.4.0
enzyme@2.9.0
express@4.15.3
jest@20.0.4
mocha@3.4.2
occasion@2.0.0
peer-tester@1.0.6
prop-types@15.5.10
react@15.6.1
-- 📓️ Note: peerDependencies for testing the latest version as well.
-- Or does this need to stay @15.4.1

JeromeFitz committed:
[fix] #38 dehumanize-date, +packages, +npm@5

🚧️ #38 dehumanize-date
This broke the buuild (seemingly) due to the package.json needing to be
updated to the latest.
At least, that is what had all the tests pass and the build pass.

🐻️ npm@5
+ package-lock.json
+ change `prepublish` to `prepare`
-- 📓️ Note: `prepublishOnly` = `upload-only` if that is desired
behaviour

📦️ package.json
+ Removed `peerDependencies` and put `react` into `dependencies`
-- 📓️ Note: It was messing up the pretest + test

📦️ Packages Updated
babel-jest@20.0.3
dehumanize-date@2.4.0
enzyme@2.9.0
express@4.15.3
jest@20.0.4
mocha@3.4.2
occasion@2.0.0
peer-tester@1.0.6
prop-types@15.5.10
react@15.6.1
-- 📓️ Note: peerDependencies for testing the latest version as well.
-- Or does this need to stay @15.4.1

JeromeFitz commented:
Errant comment in regards to:
> Removed `peerDependencies` and put `react` into `dependencies`

I neglected to take it out of the running commit message I had. 🤦‍♂️

ForbesLindesay commented:
Can you delete and .gitignore the package-lock.json file please.

ForbesLindesay commented on package.json:
```diff
@@ -11,30 +11,32 @@
+    "occasion": "2.0.0",
+    "prop-types": "^15.5.10",
+    "react-dom": "^15.6.1",
+    "react-test-renderer": "^15.6.1"
```
react-dom and react-test-render need to be devDependencies, not dependencies (and react needs to remain a peer-dependency & devDependency.

JeromeFitz replied:
Thanks. With the removal of `package-lock.json` I made sure all packages have `^` in front of them again for semver instead of lockdown.

ForbesLindesay commented:
updating peerDependencies is fine, although not strictly necessary.

JeromeFitz commented:
You got it. I'll have an updated PR for review.

JeromeFitz committed:
[refactor] Updated package.json per @ForbesLindesay

📦️ package.json
+ Added back `^` for semver to packages
- package-lock.json removed
+ package-lock.json added to .gitignore
+ moved `react-dom` and `react-test-renderer` to `devDependencies`

JeromeFitz committed:
[chore] #40 📦️ Package Updated: enzyme@2.9.1

This continues to fix the build issue being identified.
